src/stronghold/mod.rs

```diff
@@ -135,40 +134,54 @@ impl StrongholdAdapterBuilder {
     ///
     /// [`password()`]: Self::password()
     /// [`timeout()`]: Self::timeout()
-    pub fn try_build(mut self) -> Result<StrongholdAdapter> {
+    pub fn try_build(mut self, snapshot_path: PathBuf) -> Result<StrongholdAdapter> {
         // In any case, Stronghold - as a necessary component - needs to be present at this point.
         let stronghold = if let Some(stronghold) = self.stronghold {
             stronghold
         } else {
-            let system = ActorSystem::new()?;
-            let client_path = PRIVATE_DATA_CLIENT_PATH.to_vec();
-            let options = Vec::new();
-
-            Arc::new(Mutex::new(Stronghold::init_stronghold_system(
-                system,
-                client_path,
-                options,
-            )))
+            Stronghold::default()
         };
 
+        if let Some(key_provider) = &self.key_provider {
+            let result = stronghold.load_client_from_snapshot(
+                PRIVATE_DATA_CLIENT_PATH,
+                key_provider,
+                &SnapshotPath::from_path(&snapshot_path),
+            );
+            if let Err(iota_stronghold::ClientError::SnapshotFileMissing(_)) = result {
+                stronghold.create_client(PRIVATE_DATA_CLIENT_PATH)?;
+                stronghold.write_client(PRIVATE_DATA_CLIENT_PATH)?;
```

Comment:
`wirite_client` at this point would not be necessary to call, since `commit` will already write the just created client.